pallets/relayer/src/lib.rs

```diff
@@ -0,0 +1,396 @@
+        /// # Arguments
+        /// - `user_key` the user key to remove the subsidy from.
+        /// - `paying_key` the paying key that was subsidising the `user_key`.
+        ///
+        /// # Errors
+        /// - `NotAuthorizedForUserKey` if `origin` is not authorized to remove the subsidy for the `user_key`.
+        /// - `NoPayingKey` if the `user_key` doesn't have a `paying_key`.
+        /// - `NotPayingKey` if the `paying_key` doesn't match the current `paying_key`.
+        /// - `UnauthorizedCaller` if `origin` is not authorized to call this extrinsic.
+        ///
+        /// # Permissions
+        /// * Relayer
+        #[weight = <T as Config>::WeightInfo::remove_paying_key()]
+        pub fn remove_paying_key(origin, user_key: T::AccountId, paying_key: T::AccountId) -> DispatchResult {
+            Self::base_remove_paying_key(origin, user_key, paying_key)
+        }
+
+        /// Updates the available POLYX for a `user_key`.
+        ///
+        /// # Arguments
+        /// - `user_key` the user key to remove the subsidy from.
+        /// - `polyx_limit` the amount of POLYX available for subsidising the `user_key`.
+        ///
+        /// # Errors
+        /// - `NoPayingKey` if the `user_key` doesn't have a `paying_key`.
+        /// - `NotPayingKey` if `origin` doesn't match the current `paying_key`.
+        /// - `UnauthorizedCaller` if `origin` is not authorized to call this extrinsic.
+        ///
+        /// # Permissions
+        /// * Relayer
+        #[weight = <T as Config>::WeightInfo::update_polyx_limit()]
+        pub fn update_polyx_limit(origin, user_key: T::AccountId, polyx_limit: T::Balance) -> DispatchResult {
+            Self::base_update_polyx_limit(origin, user_key, polyx_limit)
+        }
+
+    }
+}
+
+decl_error! {
+    pub enum Error for Module<T: Config> {
+        /// The `user_key` is not attached to a CDD'd identity.
+        UserKeyCddMissing,
+        /// The `user_key` is not attached to a CDD'd identity.
+        PayingKeyCddMissing,
+        /// The `user_key` already has a `paying_key`.
+        AlreadyHasPayingKey,
+        /// The `user_key` doesn't have a `paying_key`.
+        NoPayingKey,
+        /// The `user_key` has a different `paying_key`.
+        NotPayingKey,
+        /// The signer not authorized for `paying_key`.
+        NotAuthorizedForPayingKey,
+        /// The signer not authorized for `user_key`.
+        NotAuthorizedForUserKey,
+    }
+}
+
+impl<T: Config> Module<T> {
+    fn base_set_paying_key(origin: T::Origin, user_key: T::AccountId) -> DispatchResult {
+        let PermissionedCallOriginData {
+            sender: paying_key,
+            primary_did: paying_did,
+            ..
+        } = <Identity<T>>::ensure_origin_call_permissions(origin)?;
+
+        // Create authorization for setting the `paying_key` to the `user_key`, with 0 `polyx_limit`
+        Self::unsafe_add_auth_for_paying_key(paying_did, user_key, paying_key, 0u128.into());
```

Comment:
```suggestion
        Self::unverified_add_auth_for_paying_key(paying_did, user_key, paying_key, 0u128.into());
```
(only `unsafe fn`s should be called "unsafe" or "unchecked")

Reply:
I agree that `unsafe_` isn't the best prefix for these functions, but it is what was used for many functions in the other pallets.

external-agents uses `unchecked_`.  Only the assets uses `unverified_`.

Which should we standardise on?

Reply:
Yeah it's a bit of a mess. We should standardize on `unverified_`. There's a team agreement to that effect already. :)